### DISCUSSION TITLE
How to deal with Electricity Market Directive (EU 2019/944)

### DISCUSSION THREAD

**Discussion by Xnygra** [dingo35/SmartEVSE-3.5#158] [Ideas]
In accordance with the EU directive, all EU countries are required to enforce the installation of smart meters and implement variable tariffs to encourage customers to spread their electricity usage more evenly throughout the day, making smarter use of the grid.

Each country, and in some cases regions within countries, is free to adopt its own tariff model. We are already starting to see requests for SmartEVSE to support specific tariff models tailored to individual countries. However, this approach will become unsustainable in the long run if SmartEVSE needs to include native code for every country's or region's specific model.

My suggestion—though I am not a coder, so I’m unsure of its feasibility—is to define a strict API-based integration that adjusts the maximum charging amperage. Developers from different countries could then create region-specific logic as separate "plug-ins." These plug-ins could be selected from a dropdown menu on the web interface's update page.

This approach would allow the regional logic to remain separate from the core SmartEVSE code, simplifying updates and maintenance.

What are your thoughts in this?

**Comment by dingo35** [comment]
I'd rather wait and see how this develops, and depending on how far apart or how close together the different countries wander, choose a technical solution.

It might be we go a plugin way, or just the API way, or simple parameter coding ... it all depends on how the actual policies will develop.
